support keys in env vars

luizarakaki opened the pull request:
# Support Environment Variables for API Keys
### Overview
This PR adds support for using environment variables to define API keys in the configuration file, enhancing security and flexibility for deployment environments.
### Changes
- Modified API key validation to support environment variable substitution in the key field
- Added runtime validation for API keys from environment variables to ensure proper formatting
- Refactored tests to properly validate environment variable integration
- All API keys (including those from environment variables) still must follow the required format:
  - Must be between 11-254 characters
  - Must start with mb_
  - Must match the pattern mb_[A-Za-z0-9+/=]+
### Usage
```
api-keys:
  - name: "ENV API Key"
    key: "{{env API_KEY_FROM_ENV}}"
    creator: "admin@example.com"
    group: "admin"
```

### Testing
- Creation of API keys from environment variables
- Validation of API key format from environment variables
- Error handling for improperly formatted environment variable values

# Skip keys if they exist
To guarantee that it is possible to keep the same yaml file on each instance initialization, if there is a key in the instance and the config file has a key with the same name, the creation is skipped.

Commit:
support keys in env vars

Commit:
add tests

Commit:
fix ref

Commit:
fix p

Commit:
fix env var

Commit:
refactor tests into smaller tests

Commit:
skip key if it already exists

Commit:
remove unused test

Commit:
fix binding

johnswanson commented on enterprise/backend/test/metabase_enterprise/advanced_config/file/api_keys_test.clj:
```diff
@@ -31,125 +31,214 @@
-      (testing "should fail if creator doesn't exist"
-        (try
+      (try
+        (write-config!
```
I should have noticed this before, but what's the difference between `write-config!` with the map and binding `config.file/*config*` to the map? Why do you need both?

luizarakaki replied:
We don't :/

Yeah write config creates a physical file while binding does in memory and takes precedence, so I'm keeping only the binding

Commit:
remove write-config

johnswanson committed:
rework tests slightly

- delete `cleanup-config!` entirely

- remove duplicate `mt/with-premium-features` (it's in the fixture)

- use `mt/fetch-user` to use existing test users instead of
`with/temp`-ing users

johnswanson reviewed (approved):
I think I can still approve this, I only modified the test bits :)

Commit:
remove unused deps
